app/models/user.rb

```diff
@@ -222,6 +222,18 @@ def post_successful_login
     User.current = self
   end
 
+  def set_locations_and_organizations_from_usergroups(usergroups)
+    new_locations = usergroups.map(&:locations).flatten - locations
```

Comment:
Use `Taxonomy.enabled_taxonomies` and do this in a loop or something, it's literally the same code for both taxonomies, and one or the other might not be enabled.